### PR TITLE
libpod: fix 'podman kube generate' on FreeBSD

### PR DESCRIPTION
**Pull request by dfr**
On FreeBSD, config.Spec.Linux and config.Spec.Process.Capabilities are nil pointers so this just avoids dereferencing these pointers in that case.

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
/approve
LGTM

File: libpod/kube.go [hunk 1]
```diff
@@ -1153,7 +1153,7 @@ func generateKubeSecurityContext(c *Container) (*v1.SecurityContext, bool, error
 	allowPrivEscalation := !c.config.Spec.Process.NoNewPrivileges
 
 	var capabilities *v1.Capabilities
-	if !privileged {
+	if !privileged && c.config.Spec.Process.Capabilities != nil {
 		// Running privileged adds all caps.
 		capabilities = c.capAddDrop(c.config.Spec.Process.Capabilities)
```
Comment (rhatdan) [inline libpod/kube.go:1158]: Should we move the check to c.capAddDrop and just check for nil there?

Comment (dfr) [reply libpod/kube.go:1158]: That also works and is probably more readable.

File: libpod/kube.go [hunk 2]
```diff
@@ -1134,16 +1134,19 @@ func determineCapAddDropFromCapabilities(defaultCaps, containerCaps []string) *v
...
-	containerCaps := make([]string, 0, len(caps.Ambient)+len(caps.Bounding)+len(caps.Effective)+len(caps.Inheritable)+len(caps.Permitted))
-	containerCaps = append(containerCaps, caps.Ambient...)
-	containerCaps = append(containerCaps, caps.Bounding...)
-	containerCaps = append(containerCaps, caps.Effective...)
-	containerCaps = append(containerCaps, caps.Inheritable...)
-	containerCaps = append(containerCaps, caps.Permitted...)
-
-	calculatedCaps := determineCapAddDropFromCapabilities(c.runtime.config.Containers.DefaultCapabilities, containerCaps)
-	return calculatedCaps
+	if caps != nil {
```
Comment (rhatdan) [inline libpod/kube.go:1137]: Nit: this would be better if it was

if caps == nil {
      return nil
}
...

Comment (dfr) [reply libpod/kube.go]: Good idea - I also added a comment to explain why its there.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Review by vrothberg — APPROVED** [review]
/lgtm
Thanks!

**Pull request merged**